### DISCUSSION TITLE
Can't load pipeline, no module named srsly.ujson.ujson

### DISCUSSION THREAD

**Discussion by diggymacd** [explosion/spaCy#8199] [Help: Installation] [answered]
spaCy appears to install ok (on Windows 10), but when I try to download encore_web_sm (or any other), I get "ModuleNotFoundError: No module named: srsly.ujason.ujason'.  I've tried installing with both Conda and Pip, using Python 3.8 and 3.9, both with and without a virtual environment.  I see srsly module in the list of dependencies.  I've tried updating srsly.  Same error whatever I do.  My install was 'conda install -c conda-forge spacy'.   What am I doing wrong??  Thanks

**Comment by polm** [comment, marked as answer]
Sorry you're having trouble with install. I've never heard of an issue like that before. Can you clarify some details?

1. Can you paste the full error traceback? 
2. If you uninstall and reinstall srsly, can you run `import srsly.ujson`?

**Comment by diggymacd** [comment]
Thanks for getting back to me so quickly.  I tried uninstalling and 
reinstall srsly, and trying to import srsly.ujson gives me the same 
error.

I've attached a text file capturing the entire process of installing 
spaCy and then attempting to load the pipeline.

Cheers,
Doug

------ Original Message ------
From: "polm" ***@***.***>
To: "explosion/spaCy" ***@***.***>
Cc: "diggymacd" ***@***.***>; "Author" 
***@***.***>
Sent: 5/25/2021 10:29:02 PM
Subject: Re: [explosion/spaCy] Can't load pipeline (#8199)

**Reply by polm** [reply]
Thanks for the detailed info. 

If you look at the output, you can see that your spaCy and Thinc installs are in conda, but your srsly is in your non-conda Python env. 

    File "C:\Users\Doug\miniconda3\envs\natlang\lib\site-packages\spacy\__init__.py", line 10, in <module>
    vs
    File "C:\Users\Doug\AppData\Local\Programs\Python\Python38\Lib\site-packages\srsly\__init__.py", line 1, in <module>

Can you try uninstalling srsly with pip and then installing it with conda directly? 

I don't generally use conda, but mixing conda and pip can cause weird issues like this.

**Comment by diggymacd** [comment]
Hi:
Yes, that solved the problem!  Thanks a lot for your help and rapid 
response.

Best regards,
Doug

------ Original Message ------
From: "polm" ***@***.***>
To: "explosion/spaCy" ***@***.***>
Cc: "diggymacd" ***@***.***>; "Author" 
***@***.***>
Sent: 5/26/2021 10:22:07 PM
Subject: Re: [explosion/spaCy] Can't load pipeline (#8199)

**Comment by cuhe-la** [comment]
In general --no-binary option for pip helps me for this kind of issues:

`python -m pip install srsly --no-binary=:all:`
or
`pip install srsly --no-binary=:all:`

Pip compiles wheel rather than download ready one from repository.

More information about --no-binary option: https://pip.pypa.io/en/latest/cli/pip_install/#cmdoption-no-binary
